### PR TITLE
perf: Improve spiking performance

### LINKED ISSUE #18: Improve spiking brain performance
The current spiking NN is not performing at all.

### PR DESCRIPTION
**Pull request by chrisjz**
Changes:

- Replace previous spiking NN brain with surrogate gradient descent.
- Add new top benchmarks for spiking agent in all environment types

Resolves #18 

## Summary by CodeRabbit

* **New Features**
  * Spiking networks now use surrogate-gradient policy-gradient learning; added population coding, temporal modulation, and richer exploration controls.
  * Expanded hyperparameter surface (LR schedules, entropy, clipping, intra-episode/batch updates) and new example experiment configurations.

* **Documentation**
  * Updated README, design/proposal/spec docs and experiment logbooks with methodology, results and migration notes.
  * Leaderboards and benchmark entries expanded with multiple new spiking results across categories.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: BENCHMARKS.md [hunk 1]
```diff
@@ -194,6 +196,7 @@ This creates a benchmark file in `benchmarks/<category>/<experiment-id>.json`.
 |---|---|---|---|---|---|---|---|
 | ✓ mlp | 0.822 | 100% | 181 | 20 | 0.000 | @chrisjz | 2025-11-27 |
 | ✓ mlp | 0.776 | 100% | 240 | 20 | 0.000 | @chrisjz | 2025-11-23 |
+| ✓ spiking | 0.733 | 100% | 267 | 22 | 0.000 | @chrisjz | 2025-12-21 |
```
[the remark this answers is not in the document]
Comment (chrisjz) [reply BENCHMARKS.md]: I will allow these in our case as we will merge related changes into `master` shortly.

File: configs/examples/spiking_foraging_small.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    update_frequency: 10
+
+    # Exploration settings
+    exploration_logit_clamp: 6.0
+    exploration_logit_clamp_final: 10.0
+    exploration_noise_std: 0.1
+    exploration_noise_std_final: 0.0
+    exploration_temperature: 1.0
+    exploration_temperature_final: 1.0
+    exploration_decay_episodes: 0
```
[the remark this answers is not in the document]
Comment (chrisjz) [reply configs/examples/spiking_foraging_small.yml]: The final values are still used, they are used in calculations outside the exploration decay episode checks.

File: docs/experiments/logbooks/003-spiking-brain-optimization.md [hunk 3]
```diff
@@ -0,0 +1,1470 @@
...
+entropy_beta: 0.01 → 0.05        # 5x stronger exploration
+```
+
+**Results**: Gradient norms dropped from infinity to 0.01-1.0 (healthy range)
+
+#### Phase 3: Performance Inconsistency (Attempts 7-12)
+**Problem**: High variance across sessions (25%-65% success rate)
+
+**Diagnosis**:
+| Session | Success Rate | Pattern |
```
[the remark this answers is not in the document]
Comment (chrisjz) [reply docs/experiments/logbooks/003-spiking-brain-optimization.md]: Should be addressed in latest commit
